adapters/between/between.go

```diff
@@ -3,169 +3,223 @@ package between
-	"github.com/prebid/prebid-server/macros"
 	"github.com/prebid/prebid-server/openrtb_ext"
-
-	"net/http"
-	"text/template"
 )
 
 type BetweenAdapter struct {
 	EndpointTemplate template.Template
 }
 
-func (a *BetweenAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-
-	errs := make([]error, 0, len(request.Imp))
-	headers := http.Header{
-		"Content-Type": {"application/json"},
-		"Accept":       {"application/json"},
-	}
-
-	// Pull the host info from the bidder params.
-	reqImps, errs := splitImpressions(request.Imp)
+const (
+	defaultBidfloor = 0.00001 // BetweenSSP requires bidfloor > 0. If BidFloor of openrtb_ext.ExtImpBetween is zero, set it to defaultBidFloor value, see addImpProps
+)
 
-	if len(reqImps) == 0 {
+func (a *BetweenAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	if len(request.Imp) == 0 {
 		return nil, []error{&errortypes.BadInput{
 			Message: fmt.Sprintf("No valid Imps in Bid Request"),
 		}}
 	}
-
-	requests := []*adapters.RequestData{}
-
-	for reqExt, reqImp := range reqImps {
-		request.Imp = reqImp
-		reqJson, err := json.Marshal(request)
-
-		if err != nil {
-			errs = append(errs, err)
-			continue
+	ext, errors := preprocess(request)
+	if errors != nil && len(errors) > 0 {
+		return nil, errors
+	}
+	endpoint, err := a.buildEndpointURL(ext)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Failed to build endpoint URL: %s", err),
+		}}
+	}
+	data, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error in packaging request to JSON"),
+		}}
+	}
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		if request.Device.DNT != nil {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
 		}
+	}
+	if request.Site != nil {
+		addHeaderIfNonEmpty(headers, "Referer", request.Site.Page)
+	}
 
-		urlParams := macros.EndpointTemplateParams{Host: reqExt.Host}
-		url, err := macros.ResolveMacros(a.EndpointTemplate, urlParams)
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     endpoint,
+		Body:    data,
+		Headers: headers,
+	}}, errors
+}
 
-		if err != nil {
-			errs = append(errs, err)
-			continue
+func unpackImpExt(imp *openrtb.Imp) (*openrtb_ext.ExtImpBetween, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ignoring BidRequest: no valid Imps",
```

Comment:
Please consider changing this message to saying `Invalid Bidder ext` or something along those lines. Similar to what you have below on L83

Reply:
fixed